server/entity/splashable.go

```diff
@@ -67,24 +82,26 @@ func (s *splashable) splash(e world.Entity, w *world.World, pos mgl64.Vec3, res
 				splashed.AddEffect(effect.New(eff.Type().(effect.LastingType), eff.Level(), dur))
 			}
 		}
-	} else if s.t == potion.Water() {
-		switch result := res.(type) {
-		case trace.BlockResult:
-			blockPos := result.BlockPosition().Side(result.Face())
-			if w.Block(blockPos) == fire() {
-				w.SetBlock(blockPos, nil, nil)
+	}
+	switch result := res.(type) {
+	case trace.BlockResult:
+		pos := result.BlockPosition().Side(res.Face())
+		if b, ok := w.Block(pos).(SplashableBlock); ok {
+			if _, ok := b.Model().(model.Empty); ok {
```

Comment:
I don't fully understand this check for `model.Empty`. Why is this necessary? And why does it `break` afterwards?

Reply:
The Model check isn't neccesary but the break is required to stop the block underneath from getting splashed. I could ofcourse use an else statement so that I don't have to use a return. I'm not really sure which would be more idomatic though.

Reply:
Behavior matches vanilla Minecraft when splashing against a campfire with normal fire in front of it.